### PR TITLE
Replace 'ERR_EXPLAIN' with 'ERR_FAIL_*_MSG' in rest of 'modules/'

### PR DESCRIPTION
**Pull request by profan**
Went through the rest of 'modules/' to replace most usages remaining of `ERR_EXPLAIN` in there, however left with a few places where it was again disjointed from the error slightly, first one:

https://github.com/godotengine/godot/blob/3418f76a9eab9f496e5b26310bd3bc1125b8119b/modules/etc/texture_loader_pkm.cpp#L66-L69

as well as a second one: https://github.com/godotengine/godot/blob/3418f76a9eab9f496e5b26310bd3bc1125b8119b/modules/hdr/image_loader_hdr.cpp#L47-L51

and lastly one `ERR_EXPLAINC` that was in use inside the xatlas_unwrap module:
https://github.com/godotengine/godot/blob/3418f76a9eab9f496e5b26310bd3bc1125b8119b/modules/xatlas_unwrap/register_types.cpp#L65-L66

In the last case, without importing godot's String, the normal error macro couldn't be used directly, so that's why I decided to just put this up for debate.

Part of #31207.

### CODE REVIEW

File: modules/assimp/editor_scene_importer_assimp.cpp [hunk 1]
```diff
@@ -779,8 +777,7 @@ Ref<Texture> EditorSceneImporterAssimp::_load_texture(ImportState &state, String
 				t->set_storage(ImageTexture::STORAGE_COMPRESS_LOSSY);
 				return t;
 			} else if (tex->CheckFormat("dds")) {
-				ERR_EXPLAIN("Open Asset Import: Embedded dds not implemented");
-				ERR_FAIL_COND_V(true, Ref<Texture>());
+				ERR_FAIL_COND_V_MSG(true, Ref<Texture>(), "Open Asset Import: Embedded dds not implemented.");
```
Comment (akien-mga) [inline modules/assimp/editor_scene_importer_assimp.cpp]: This should simply be `ERR_FAIL_V_MSG`, no need for `COND` if it's `true`.

File: modules/assimp/editor_scene_importer_assimp.cpp [hunk 2]
```diff
@@ -150,8 +150,7 @@ Node *EditorSceneImporterAssimp::import_scene(const String &p_path, uint32_t p_f
 								 0;
 	const aiScene *scene = importer.ReadFile(s_path.c_str(),
 			post_process_Steps);
-	ERR_EXPLAIN(String("Open Asset Import failed to open: ") + String(importer.GetErrorString()));
-	ERR_FAIL_COND_V(scene == NULL, NULL);
+	ERR_FAIL_COND_V_MSG(scene == NULL, NULL, String("Open Asset Import failed to open: ") + String(importer.GetErrorString()) + ".");
```
Comment (akien-mga) [inline modules/assimp/editor_scene_importer_assimp.cpp]: Could be `ERR_FAIL_NULL_V_MSG(scene, ...)`, though it's also fine to keep as is in this PR. A regex could likely fix current occurrences of `ERR_FAIL_COND.*([A-Za-z0-9_]* == NULL`.

File: modules/bmp/image_loader_bmp.cpp [hunk 3]
```diff
@@ -241,9 +241,8 @@ Error ImageLoaderBMP::load_image(Ref<Image> p_image, FileAccess *f,
 				case BI_CMYKRLE8:
 				case BI_CMYKRLE4: {
 					// Stop parsing
-					ERR_EXPLAIN("Compressed BMP files are not supported: " + f->get_path());
 					f->close();
-					ERR_FAIL_V(ERR_UNAVAILABLE);
+					ERR_FAIL_V_MSG(ERR_UNAVAILABLE, "Compressed BMP files are not supported: " + f->get_path() + ".");
```
Comment (akien-mga) [inline modules/bmp/image_loader_bmp.cpp]: Might crash here, I think `f->close()` nullifies the pointer.

File: modules/dds/texture_loader_dds.cpp [hunk 4]
```diff
@@ -128,8 +127,7 @@ RES ResourceFormatDDS::load(const String &p_path, const String &p_original_path,
 
 	if (magic != DDS_MAGIC || hsize != 124 || !(flags & DDSD_PIXELFORMAT) || !(flags & DDSD_CAPS)) {
 
-		ERR_EXPLAIN("Invalid or Unsupported DDS texture file: " + p_path);
-		ERR_FAIL_V(RES());
+		ERR_FAIL_V_MSG(RES(), "Invalid or Unsupported DDS texture file: " + p_path + ".");
```
Comment (akien-mga) [inline modules/dds/texture_loader_dds.cpp]: Nitpick, but I'd make "Unsupported" lowercase here and in the next occurrence.

File: modules/enet/networked_multiplayer_enet.cpp [hunk 5]
```diff
@@ -543,10 +543,7 @@ Error NetworkedMultiplayerENet::put_packet(const uint8_t *p_buffer, int p_buffer
 	if (target_peer != 0) {
 
 		E = peer_map.find(ABS(target_peer));
-		if (!E) {
-			ERR_EXPLAIN("Invalid Target Peer: " + itos(target_peer));
-			ERR_FAIL_V(ERR_INVALID_PARAMETER);
-		}
+		ERR_FAIL_COND_V_MSG(!E, ERR_INVALID_PARAMETER, "Invalid Target Peer: " + itos(target_peer) + ".");
```
Comment (akien-mga) [inline modules/enet/networked_multiplayer_enet.cpp]: I'd make "Target Peer" lowercase.

File: modules/opus/audio_stream_opus.cpp [hunk 6]
```diff
@@ -287,8 +287,7 @@ int AudioStreamPlaybackOpus::mix(int16_t *p_buffer, int p_frames) {
 		int ret = op_read(opus_file, (opus_int16 *)p_buffer, todo * stream_channels, &current_section);
 		if (ret < 0) {
 			playing = false;
-			ERR_EXPLAIN("Error reading Opus File: " + file);
-			ERR_BREAK(ret < 0);
+			ERR_BREAK_MSG(ret < 0, "Error reading Opus File: " + file + ".");
```
Comment (akien-mga) [inline modules/opus/audio_stream_opus.cpp]: Lowercase "File".

File: modules/squish/image_compress_squish.cpp [hunk 7]
```diff
@@ -57,8 +57,7 @@ void image_decompress_squish(Image *p_image) {
 	} else if (p_image->get_format() == Image::FORMAT_RGTC_RG) {
 		squish_flags = squish::kBc5;
 	} else {
-		ERR_EXPLAIN("Squish: Can't decompress unknown format: " + itos(p_image->get_format()));
-		ERR_FAIL_COND(true);
+		ERR_FAIL_COND_MSG(true, "Squish: Can't decompress unknown format: " + itos(p_image->get_format()) + ".");
```
Comment (akien-mga) [inline modules/squish/image_compress_squish.cpp]: Should be `ERR_FAIL_MSG`.

File: modules/vorbis/audio_stream_ogg_vorbis.cpp [hunk 8]
```diff
@@ -116,8 +116,7 @@ int AudioStreamPlaybackOGGVorbis::mix(int16_t *p_buffer, int p_frames) {
 		if (ret < 0) {
 
 			playing = false;
-			ERR_EXPLAIN("Error reading OGG Vorbis File: " + file);
-			ERR_BREAK(ret < 0);
+			ERR_BREAK_MSG(ret < 0, "Error reading OGG Vorbis File: " + file + ".");
```
Comment (akien-mga) [inline modules/vorbis/audio_stream_ogg_vorbis.cpp]: Lowercase "file" here too.

File: modules/webrtc/webrtc_multiplayer.cpp [hunk 9]
```diff
@@ -321,10 +321,8 @@ Error WebRTCMultiplayer::put_packet(const uint8_t *p_buffer, int p_buffer_size)
 	if (target_peer > 0) {
 
 		E = peer_map.find(target_peer);
-		if (!E) {
-			ERR_EXPLAIN("Invalid Target Peer: " + itos(target_peer));
-			ERR_FAIL_V(ERR_INVALID_PARAMETER);
-		}
+		ERR_FAIL_COND_V_MSG(!E, ERR_INVALID_PARAMETER, "Invalid Target Peer: " + itos(target_peer) + ".");
```
Comment (akien-mga) [inline modules/webrtc/webrtc_multiplayer.cpp]: Lowercase "target peer".

File: modules/websocket/wsl_server.cpp [hunk 10]
```diff
@@ -45,29 +45,18 @@ WSLServer::PendingPeer::PendingPeer() {
 bool WSLServer::PendingPeer::_parse_request(const PoolStringArray p_protocols) {
 	Vector<String> psa = String((char *)req_buf).split("\r\n");
 	int len = psa.size();
-	if (len < 4) {
-		ERR_EXPLAIN("Not enough response headers.");
-		ERR_FAIL_V(false);
-	}
+	ERR_FAIL_COND_V_MSG(len < 4, false, "Not enough response headers.");
```
Comment (akien-mga) [inline modules/websocket/wsl_server.cpp]: Maybe use the improved message you wrote for the other class:
```
ERR_FAIL_COND_V_MSG(len < 4, false, "Not enough response headers, got: " + itos(len) + ", expected >= 4.");
```

**Comment by akien-mga** [comment]
Left some nitpicks, but otherwise good to merge.

File: modules/bmp/image_loader_bmp.cpp (hunk 3 above)
Comment (profan) [reply modules/bmp/image_loader_bmp.cpp]: Oh yeah, fair, get the path and separate out into var first i guess?

Comment (akien-mga) [reply modules/bmp/image_loader_bmp.cpp]: Yes, you can save the path first and use it here. This `case` is an error condition in itself already so the performance impact is not a problem.

File: modules/assimp/editor_scene_importer_assimp.cpp (hunk 2 above)
Comment (profan) [reply modules/assimp/editor_scene_importer_assimp.cpp]: Keeping as it is for now at least

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**